README.md
admin_scripts
=============

Scripts de infra que desenvolvi


bkp_database/Changes
1.0  Tue Oct 9 15:04:33 2012.
	- Criação do script

2.0  Dom Dec 23 21:28:06 2012.
  - Documentação POD



restore.sh
#!/bin/bash
# =======================================================================================
# restore.sh
# =======================================================================================
# Restaura uma base em SGBD PostgreSQL a partir de um dump SQL compactado com bzip2
# @author: Lucas Pereira
# @since: 23/01/2012
# @param string: IP ou nome do servidor PostgreSQL
# @param integer: porta do servidor
# @param string: usuário do SGBD 
# @param string: senha do usuário
# @param string: nome da base a ser criada
# @param string: arquivo do dump (caminho)
#========================================================================================


# =====================================================================
# CONSTANTES
# =====================================================================
readonly TRUE=1
readonly FALSE=0
# =====================================================================

declare -A CONN
CMD_BUNZIP="/bin/bunzip2"
CMD_PSQL="/usr/lib/postgresql/9.1/bin/psql"
CMD_RESTORE="/usr/lib/postgresql/9.1/bin/pg_restore"
NOME_BASE=$5
ARQ_DUMP=$6
CMD_EXECUTAR=$7

# =====================================================================
# testaerro
# =====================================================================
# Verifica status de comando. Caso o status seja diferente de zero, A função poderá disparar um erro, ou abortar a execução do programa (caso $2 esteja configurado)
# @param integer: status do comando executado
# @param integer: define erro fatal (aborta execução em caso de erro)
# =====================================================================

function testaerro () {
  if [ $1 -ne 0 ] && [ $2 -eq $TRUE ]
  then
    echo -ne "[ FATAL ]\n" >&2
    exit $1
  elif [ $1 -ne 0 ]
  then
    echo -ne "[ ERRO ]\n" >&2
    echo "$1"
  elif [ $1 -eq 0 ]
  then
    echo -ne "[ OK ]\n"
  fi
}

# =====================================================================



# =====================================================================
# executa_SQL
# =====================================================================
# Executa comando SQL na base inicializada com ini_db
# @param string: Comando SQL a ser executado.
# =====================================================================

function executa_sql () { 
  SQL="$1"
  DISCARD_STDOUT=$2
  if [ -z $PGPASSWORD ] && [ -z ${CONN[senha]} ]
  then
    export PGPASSWORD=${CONN[senha]}
  fi
  if [ $DISCARD_STDOUT -eq $TRUE ]
  then
    $CMD_PSQL -U ${CONN[usuario]} -h ${CONN[host]} -p ${CONN[porta]} ${CONN[base]} -t -A -c "$SQL" >/dev/null
  elif [ $DISCARD_STDOUT -eq $FALSE ]
  then
    $CMD_PSQL -U ${CONN[usuario]} -h ${CONN[host]} -p ${CONN[porta]} ${CONN[base]} -t -A -c "$SQL"
  fi
  testaerro $? 1
  unset PGPASSWORD
}

# =====================================================================


#========================================================================================
# INÍCIO DO PROGRAMA
#========================================================================================
CMD_BUNZIP="/bin/bunzip2"
CMD_PSQL="/usr/lib/postgresql/9.1/bin/psql"
CMD_RESTORE="/usr/lib/postgresql/9.1/bin/pg_restore"


NOME_BASE=$5
ARQ_DUMP=$6
CMD_EXECUTAR=$7
STATUS=0


#========================================================================================
# RESTAURAÇÃO DA BASE
#========================================================================================
CONN=([host]=$1
        [porta]=$2
        [usuario]=$3
        [senha]=$4
        [base]="postgres") 
executa_sql 'CREATE DATABASE "'$NOME_BASE'";' $FALSE

CONN=([host]=$1
      [porta]=$2
      [usuario]=$3
      [senha]=$4
      [base]=$NOME_BASE)

[ ! -z $PGPASSWORD -a -z ${CONN[senha]} ] || export PGPASSWORD=${CONN[senha]}
if [ "$(file $ARQ_DUMP | cut -d ':' -f 2 | cut -d ' ' -f 2)" = "bzip2" ] 
then
  $CMD_BUNZIP -c $ARQ_DUMP | psql -U ${CONN[usuario]} -h ${CONN[host]} -p ${CONN[porta]} ${CONN[base]}
  STATUS=$?
elif [ "$(file $ARQ_DUMP | cut -d ':' -f 2 | cut -d ' ' -f 2)" = "PostgreSQL" ]
then
  $CMD_RESTORE -U ${CONN[usuario]} -h ${CONN[host]} -p ${CONN[porta]} -d ${CONN[base]} -L $ARQ_DUMP.list $ARQ_DUMP -v
  STATUS=$?
fi

if [ $STATUS -ne 0 ]
then
  CONN=([host]=$1
        [porta]=$2
        [usuario]=$3
        [senha]=$4
        [base]="postgres")
  executa_sql "DROP DATABASE \"$NOME_BASE\";" $FALSE # Em caso de erro, dropa a base que estava sendo restaurada.
  testaerro 1 1
else
  testaerro 0 1
  $CMD_PSQL -U postgres -h ${CONN[host]} -p ${CONN[porta]} ${CONN[base]} << SQL
    SET vacuum_cost_delay = 0;
    VACUUM ANALYZE;
SQL
fi
#========================================================================================


# EXECUÇÃO DO SCRIPT PÓS-RESTAURAÇÃO
[ -z "$CMD_EXECUTAR" ] || executa_sql "$CMD_EXECUTAR" $FALSE 
exit 0


checkurl/checkurl.pl
#!/usr/bin/perl
# ====================================================================================
# checkurl.pl
# ====================================================================================
# Script de verificação da URL a ser acessada pelo Squid.
# @author: Lucas Pereira
# @since: 04-04-2012
# ====================================================================================
require 'functions.ph';
use Net::SMTP;
use IO::File;
use DBI;
use Data::Dumper qw(Dumper);
use autodie;

# ===================================================
# Tratamento de sinais
# ===================================================
$SIG{INT}  = 'IGNORE';
$SIG{TERM} = 'IGNORE';
$SIG{QUIT} = 'IGNORE';
# ===================================================

%hConfig = ();
readConf ();

$oArqLog = IO::File->new ();
$oArqLog->open ($hConfig{'log_file'}, "a");
$oArqLog->print ("[" . localtime () . "] Iniciando script de verificação de URL do Squid.\n");
$oDB     = DBI->connect ("dbi:Pg:dbname=" . $hConfig{'database_base'} . ";host=" . $hConfig{'database_host'} . ";port=" . $hConfig{'database_port'}, $hConfig{'database_user'}, $hConfig{'database_password'}); 
$|       = 1; # Descarrega buffer de entrada padrão. Esta flag vale em toda a execução do script.
@aTbl = mapTables ();
$oArqLog->close();

while ($sURL = <>) {
  chop ($sURL);
  $oArqLog->open ($hConfig{'log_file'}, "a");
  if ($sURL ne "") {
    # ===================================================
    # Processamento da URL (extração do domínio)
    # ===================================================
    @aURL    = split ( /\//, $sURL);
    $sDomain = $aURL[2];
    undef (@aURL);
    @aDomain = split (/\./, $sDomain);
    $sSuffixTbl  = pop (@aDomain);
    $sNomeTabela = "domains_" . $sSuffixTbl;
    push (@aDomain, $sSuffixTbl);
    if ($aDomain[0] =~ /^www(|[0-100])\.*/) {
      shift (@aDomain);
    }
    undef ($sDomain);
    $sDomain = join (".",@aDomain);
    undef (@aDomain);
    $oArqLog->print ("[" . localtime () . "] $sDomain\n");
    # ===================================================


    # ===================================================
    # Validação do acesso
    # ===================================================
    if (grep { $_ = $sNomeTabela } @aTbl) {
      eval {
        $sSQLDomain = "SELECT url
                         FROM $sNomeTabela
                        WHERE url = '$sDomain';";
        $oResDomain = $oDB->prepare ($sSQLDomain);
        $oResDomain->execute ();
        if ($oResDomain->rows() > 0) {
          print "OK\n";
          $oArqLog->print ("[" . localtime () ."] OK\n");
        } else {
          print "ERR\n";
          $oArqLog->print ("[" . localtime () . "] ERR\n");
        }
        $oResDomain->finish ();
      };
      if ($@) {
        $sErr  = $@;
        $oArqLog->print ("$sErr\n");
        # Envia e-mail em caso de erro na conexão com o banco de dados
        $oMail = Net::SMTP->new ($hConfig{'mail_host'});
        $oMail->mail ('squid');
        $oMail->to ($hConfig{'mail_send'});
        $oMail->data ();
        $oMail->datasend ('To: ' . $hConfig{'mail_send'});
        $oMail->datasend ("\n");
        $oMail->datasend ('Subject: Squid - [ ERRO ]');
        $oMail->datasend ("\n");
        $oMail->datasend ("==========================================================\n");
        $oMail->datasend ("E-MAIL AUTOMÁTICO DO SCRIPT DE VERIFICAÇÃO DE URL DO SQUID\n");
        $oMail->datasend ("==========================================================\n\n");
        $oMail->datasend ("OCORREU UM ERRO AO CONECTAR COM A BASE DE DADOS DE URL'S DO SQUID:\n");
        $oMail->datasend ("$sErr");
        $oMail->dataend ();
        $oMail->quit ();
      }
      # ===================================================
    } else {
      print "ERR\n"; # Se a tabela contendo o domínio não for encontrada, considera a URL válida.
      $oArqLog->print ("[" . localtime () ."] ERR\n");
    }
  }
  $oArqLog->close ();
}

# ===================================================
# Encerrando programa
# ===================================================
$oArqLog->open ($hConfig{'log_file'}, "a");
$oArqLog->print ("[" . localtime () . "] AVISO: EOF detectado. Encerrando programa...\n");
$oResDomain->finish () if ($oResDomain);
$oDB->disconnect () if ($oDB);
$oArqLog->close ();
# ===================================================
exit 0;


bkp_database/Makefile.PL
use 5.010001;
use ExtUtils::MakeMaker;
# See lib/ExtUtils/MakeMaker.pm for details of how to influence
# the contents of the Makefile that is written.
my $DESTDIR = $ENV{DESTDIR} || '';
my $INSTALLDIRS = $ENV{INSTALLDIRS} || 'site';

WriteMakefile(
    NAME              => 'bkp_database',
    DIST_NAME         => 'bkp_database',
    VERSION_FROM      => 'bkp_database.pl', # finds $VERSION
    PREREQ_PM         => {
                           DBD::Pg     => '2.16.1',
                           XML::Simple => '2.20'
                         }, # e.g., Module::Name => 1.1
    ($] >= 5.005 ?     ## Add these new keywords supported since 5.005
      (ABSTRACT_FROM  => 'bkp_database.pl', # retrieve abstract from module
       AUTHOR         => 'Lucas Pereira <lucas.pereira@dbseller.com.br>' 
       MAN1PODS       => {
                           'doc/bkp_database.pod',
                           'doc/bkp_srv.pl'
                         }
        EXE_FILES     => [ qw(bkp_database.pl bkp_srv.pl) ]
     ) : ()),
);


bkp_database/bkp_database.pl
#!/usr/bin/perl
# =================================================================================
# bkp_database.pl
# =================================================================================
# Rotina de backup das bases através de pg_dump (cliente)
# Envia as informações ao servidor para que seja feito backups no servidor.
# Somente as bases listadas no arquivo databases terão backup.
# Pode fazer backups em vários clusters diferentes em um mesmo servidor.
#
# @author: Lucas Pereira
# @since: 22-06-2012
# =================================================================================
use DBI;
use IO::Socket::INET;
use IO::File;
use XML::Simple;
use autodie;

our $VERSION = 2.0;
# ==========================================
# Parâmetros passados por linha de comando
# ==========================================
$sHost      = $ARGV[0];                                         # Servidor onde estão as bases
$sPathArqDB = $ARGV[1];                                         # Caminho para o arquivo databases
# ==========================================

@aHost   = split (/:/, $sHost);
@aConfDB = ("port", "user", "password", "database", "version"); # Array de configurações para criar o 

$oSock   = IO::Socket::INET->new (
                                   PeerHost  => $aHost[0],
                                   PeerPort  => $aHost[1],
                                   Proto     => 'tcp',
                                   Type      => SOCK_STREAM,
                                   Timeout   => 30
                                 ) or die ($!);
$oSock->autoflush (1);
$oArqDB = IO::File->new ($sPathArqDB, "r") or die ($!);                    # Arquivo databases
$sLine  = $oArqDB->getline ();

do {
  @aLine = split (/,/, $sLine);
  $i     = 0;
  foreach $sLinha (@aLine) {
    $hConfDB{$aConfDB[$i]} = $sLinha;
    $oSock->send ("$sLinha\n");                                # Envia informações para fazer dump da base.
    sleep (2);
    $i++;
  }
  $oSock->recv ($_, 20);                                       # Recebe status do pg_dump enviado pelo servidor

  # ==========================================
  # VACUUM ANALYZE
  # ==========================================
  $oSock->send ("VACUUM ANALYZE\n");
  eval {
    $oDB = DBI->connect ("dbi:Pg:dbname=" . $hConfDB{'database'} . ";host=" . $aHost[0] . ";port=" . $hConfDB{'port'}, $hConfDB{'user'}, $hConfDB{'password'}, {
                                                                                                                                                                 RaiseError         => 1, 
                                                                                                                                                                 PrintWarn          => 1,
                                                                                                                                                                 ShowErrorStatement => 1,
                                                                                                                                                               });

    $oDB->do ("VACUUM ANALYZE;");
    $oSock->send ("OK\n");
    $oDB->disconnect ();
  };
  $oSock->send ($@ . "\n") if ($@);
  # ==========================================

  if (($sLine = $oArqDB->getline ())) {
    $oSock->send ("1\n");                                    # Informa ao servidor que existem mais bases que devem ser feitas backup.
  } else {
    $oSock->send ("0\n");                                    # Sinaliza para o servidor que não existem mais bases para backup.
  }
} while ($sLine ne "");

$oArqDB->close ();
$oSock->shutdown (2);
exit 0;


__END__
=head1 NOME

  bkp_database - Serviço de rotina de backup para bancos de dados PostgreSQL (cliente).

=head1 SINOPSE
  
  bkp_database.pl <host>:<porta> <arquivo databases>

=head1 DESCRIÇÃO

  Cliente para execução de backup lógico e rotinas de manutenção básicas (VACUUM ANALYZE) de bancos de dados PostgreSQL utilizando pg_dump.
  Pode ser utilizado tando em modo standalone, como pode utilizado juntamente com sistemas gerenciadores de backup, como o Bacula.
  
  Este script foi feito de maneira a ser possível fazer backup de bases em vários clusters em um mesmo servidor.

=head1 ARQUIVO DATABASES
  
  O arquivo databases é utilizado somente pelo cliente, e utiliza formato CSV, com as seguintes colunas:
  I<porta,usuario,senha,nome_da_base,versão_da_base>

  Este arquivo é o que torna possível fazer backup de vários clusters em um mesmo servidor.

=head1 VER TAMBÉM

  DBI
  
=head1 AUTOR

  Lucas Pereira, E<lt>lucas.pereira@dbseller.com.brE<gt>

=head1 LICENÇA

  This library is free software; you can redistribute it and/or modify
  it under the same terms as Perl itself, either Perl version 5.10.1 or,
  at your option, any later version of Perl 5 you may have available.

=cut


bkp_database/bkp_srv.pl
#!/usr/bin/perl
# =================================================================================
# bkp_srv.pl
# =================================================================================
# Rotina de backup das bases através de pg_dump (servidor)
# Faz backup das bases informadas pelo cliente.
# Pode fazer backups em vários clusters diferentes em um mesmo servidor.
# 
# @author: Lucas Pereira
# @since: 22-06-2012
# =================================================================================

use XML::Simple;
use IO::Socket::INET;
use IO::File;
use File::Copy;
use Data::Dumper qw(Dumper);

# ==========================================
# finalize
# ==========================================
# Função de callback para tratamento de
# sinais recebidos pelo processo do
# servidor.
# ==========================================
sub finalize () {
  $oLog->print ("\n[" . localtime () . "] AVISO: Encerrando servidor da rotina de backup devido à comando do administrador...\n");
  $oClient->shutdown (2) if ($oClient);
  $oSockSrv->shutdown (2) if ($oSockSrv);
  $oLog->close ();
  unlink ($hXMLConfSrv->{'daemon'}->{'config'}->{'lock_dir'}->{'value'} . "backup.pid");
  exit 0;
}
# ==========================================

die ("Uso: ./bkp_srv.pl <arquivo de configuração XML>") if ($#ARGV < 0);


# ==========================================
# Lendo configurações (XML)
# ==========================================
$sPathConfigFile = $ARGV[0];
$oXML            = XML::Simple->new ();
$hXMLConfSrv     = $oXML->XMLin ($sPathConfigFile, ValueAttr => {});
# ==========================================



$ENV{'PGHOST'}   = $hXMLConfSrv->{'bkp'}->{'config'}->{'pg_sock'}->{'value'};
# ==========================================
# Verificação de lock
# ==========================================
if (-e $hXMLConfSrv->{'daemon'}->{'config'}->{'lock_dir'}->{'value'} . "backup.pid") {
  die ("O serviço já está em execução!");
} else {
  eval {
    $oLock = IO::File->new ($hXMLConfSrv->{'daemon'}->{'config'}->{'lock_dir'}->{'value'} . "backup.pid", "w");
    $oLock->write ($$ . "\n");
    $oLock->close ();
  };
  if ($@) { 
    $sErro = $@;
    print STDERR "Não foi possível criar arquivo de lock: $@\n";
    if (! -d $hXMLConfSrv->{'daemon'}->{'config'}->{'lock_dir'}->{'value'}) {
      print STDERR "Diretório " . $hXMLConfSrv->{'daemon'}->{'config'}->{'lock_dir'}->{'value'} . " não existe.\n";
    } elsif (! -w $hXMLConfSrv->{'daemon'}->{'config'}->{'lock_dir'}->{'value'}) {
      print STDERR "Diretório " . $hXMLConfSrv->{'daemon'}->{'config'}->{'lock_dir'}->{'value'} . " não tem permissão de escrita.\n";
    }    
    exit -2;
  }
}
# ==========================================


$oLog  = IO::File->new ($hXMLConfSrv->{'daemon'}->{'config'}->{'log_dir'}->{'value'} . "backup.log", "a") or die ($!); # Arquivo de log
$oLog->autoflush (1);
$oLog->print ("[" . localtime () . "] Iniciado servidor da rotina de backup...\n");
$oLog->print ("[" . localtime () . "] Aguardando conexão...\n");
$oLog->close ();


# ==========================
# Tratamento de sinais
# ==========================
$SIG{'TERM'} = 'finalize';
$SIG{'INT'}  = 'finalize';
# ==========================


while (1) {
  $oLog->open ($hXMLConfSrv->{'daemon'}->{'config'}->{'log_dir'}->{'value'} . "backup.log", "a") or die ($!);
  $oSockSrv        = IO::Socket::INET->new (
                                             LocalPeer => $hXMLConfSrv->{'socket'}->{'config'}->{'bind_addr'}->{'value'},
                                             LocalPort => $hXMLConfSrv->{'socket'}->{'config'}->{'port'}->{'value'},
                                             Listen    => 1,
                                             Proto     => 'tcp',
                                             Type      => SOCK_STREAM
                                           );
  next if (!$oSockSrv); 
  $oSockSrv->autoflush (1);
  $oClient = $oSockSrv->accept ();                                   # O servidor está aceitando conexões
  $oLog->print ("[" . localtime () . "] Recebendo conexão de " . $oClient->peerhost () . "...\n");
  $oClient->autoflush (1);
  $oLog->print ("[" . localtime () . "] Iniciando rotina de backup...\n");

  while (1) {
    eval {

      # ==========================================
      # Processamento dos dados passados 
      # pelo cliente
      # ==========================================
      @aConfDB = ("port", "user", "password", "database", "version");  # Array com configuraçõe da base a ser feito backup.
      $oLog->print ("[" . localtime () . "] ");
      foreach $sConfDB (@aConfDB) {
        $oClient->recv ($hConfDB{$sConfDB}, 256);
        if ($sConfDB ne "password") {
          chop ($hConfDB{$sConfDB});                                     # Retira \n de cada parâmetro da configuração
          $oLog->print ("$sConfDB=" . $hConfDB{$sConfDB} . ", ");
        }
      }
      $oLog->print ("\n");
      chop ($hConfDB{'version'});                                     # Corta \n da versão. 
      # ==========================================
      

      # ==========================================
      # Backup da base
      # ==========================================
      $oLog->print ("[" . localtime () . "] Fazendo backup da base $hConfDB{'database'}... ");
      system ("touch /tmp/." . $hConfDB{'database'} . ".pgbkp");
      chown ($UID, $GID, "/tmp/." . $hConfDB{'database'} . ".pgbkp");
      system ($hXMLConfSrv->{'bkp'}->{'config'}->{'pg_exec_dir'}->{'value'} . $hConfDB{'version'} . "/bin/pg_dump -i -Fc -Z 9 -U " . $hConfDB{'user'} . " -p " . $hConfDB{'port'} . " " . $hConfDB{'database'} . "> /tmp/." . $hConfDB{'database'} . ".pgbkp");    # O arquivo do backup é gravado 
                                                                                                                                                                                                                                                                   # provisóriamente no /tmp do servidor (arquivo oculto)

      # Status do backup
      if ($? == 0) {
        $oLog->print ("[ OK ]\n");
        move ("/tmp/." . $hConfDB{'database'} . ".pgbkp", $hXMLConfSrv->{'bkp'}->{'config'}->{'bkp_dir'}->{'value'} . $hConfDB{'database'} . ".pgbkp");      # Move backup da pasta /tmp/ p/ ser disponibilizado ao Bacula.
        unlink ("/tmp/." . $hConfDB{'database'} . ".pgbkp") if (-e "/tmp/." . $hConfDB{'database'} . ".pgbkp");
        chown ($UID, $GID, $hXMLConfSrv->{'bkp'}->{'config'}->{'bkp_dir'}->{'value'} . $hConfDB{'database'} . ".pgbkp");
        chmod (0755, $hXMLConfSrv->{'bkp'}->{'config'}->{'bkp_dir'}->{'value'} . $hConfDB{'database'} . ".pgbkp");
      } else {
        $oLog->print ("[ERRO]\n");
        unlink ("/tmp/." . $hConfDB{'database'} . ".pgbkp");  # Em caso de erro no backup, apaga o backup com problema.
      }
      # ==========================================
      
      
      # ==========================================
      # VACUUM ANALYZE (executado pelo cliente)
      # ==========================================
      $oClient->send ("DUMP END\n");                          # Informa ao cliente que o dump terminou
      $oClient->recv ($_, 20);                                # Recebe sinal do cliente indicando que o mesmo está executando VACUUM ANALYZE na base.
      $oLog->print ("[" . localtime () . "] Executando VACUUM ANALYZE... ");
      $oClient->recv ($sVacuumStatus, 1024);                  # Recebe status do VACUUM para escrevê-lo no log
      chop ($sVacuumStatus);
      if ($sVacuumStatus eq "OK") {
        $oLog->print ("[ OK ]\n");
      } else {
        $oLog->print ("[ERRO]\n" . "[" . localtime () . "] $sVacuumStatus\n" );
      }
      # ==========================================


      $oClient->recv ($bNext, 2);                             # Cliente informa se há mais bases para ser feito backup ou não.
      if ($bNext eq "0\n") {                       
        $oClient->shutdown (2);                               # Caso não haja, fecha os sockets
        $oSockSrv->shutdown (2);
        undef ($oClient);
        undef ($oSockSrv);
        last;
      } elsif ($bNext ne "0\n" && $bNext ne "1\n") {
        $oLog->print ("\n[" . localtime () . "] AVISO: Recebida mensagem desconhecida do host " . $oClient->peerhost() . " ($bNext)!\n");
        $oLog->print ("\n[" . localtime () . "] Encerrando conexão com " . $oClient->peerhost() . "...\n");
        $oClient->shutdown (2);                               # Caso receba uma resposta inesperada, fecha a conexão.
        $oSockSrv->shutdown (2);
        undef ($oClient);
        undef ($oSockSrv);
        last;
      }
      undef (%hConfDB);
      
    };
    if ($@) {
      $sError = $@;
      $oLog->print ("$sError\n");
    }
  }
  $oLog->print ("[" . localtime () . "] Terminada rotina de backup!\n");
  $oLog->close ();
}

__END__

=head1 NOME

  bkp_srv - Serviço de rotina de backup para bancos de dados PostgreSQL (servidor).

=head1 SINOPSE
  
  bkp_srv.pl <arquivo_de_configuração>

=head1 DESCRIÇÃO

  Servidor para execução de backup lógico e rotinas de manutenção básicas (VACUUM ANALYZE) de bancos de dados PostgreSQL utilizando pg_dump.

=head1 ARQUIVO DE CONFIGURAÇÃO DO SERVIDOR

  O arquivo de cconfiguração do servidor é baseado em XML, o qual suporta as seguintes tags:
  =begin text

  <server>: raiz do arquivo
  <socket>: seção de configurações de socket <host, porta>. Configurações suportadas: bind_addr (endereço de bind), port (porta) e timeout.
  <daemon>: seção de configurações do serviço (locks e logs). Configurações suportadas: log_dir (diretório de log) e lock_dir (diretório de lock).
  <bkp>: seção de configurações do backup. Configurações suportadas: pg_exec_dir (diretório de executáveis do PostgreSQL) e bkp_dir (diretório onde serão armazenados os backups).
  
  =end text
  
  Cada uma das tags mencionadas acima devem conter a tag config, com o atributo "name" de valor do nome da configuração, a qual deve ser um dos nomes 
  citados nas configurações suportadas, e seu respectivo valor.


=head1 VER TAMBÉM

  man pg_dump
  
=head1 AUTOR

  Lucas Pereira, E<lt>lucas.pereira@dbseller.com.brE<gt>

=head1 LICENÇA

  This library is free software; you can redistribute it and/or modify
  it under the same terms as Perl itself, either Perl version 5.10.1 or,
  at your option, any later version of Perl 5 you may have available
=cut
